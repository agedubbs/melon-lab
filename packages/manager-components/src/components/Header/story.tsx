import { storiesOf } from '@storybook/react';
import React from 'react';
import Header from './index';

storiesOf('Components|Header', module).add('Default', () => {
  return <Header />;
});
