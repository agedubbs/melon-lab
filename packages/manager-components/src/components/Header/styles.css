.header {
  display: flex;
  align-items: center;
  height: 60px;
  background-color: var(--color-primary-dark);
  font-size: var(--font-size-s);
  min-width: 1200px;
}

.header__logo {
  display: flex;
  align-items: center;
  height: 100%;
  font-size: 0;
  line-height: 0;
  padding: 0 var(--space-s);
  border-right: 1px solid var(--color-primary-darkest);
}

.header__name {
  padding: 0 var(--space-l);
  font-weight: var(--font-weight-bold);
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.header__price {
  padding: 0 var(--space-xxl);
  overflow: hidden;
}

.header__price-value,
.header__price-desc {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.header__price-desc {
  display: block;
  font-size: var(--font-size-xs);
  color: var(--color-grey);
}

.header__price-value {
  display: block;
  font-size: var(--font-size-s);
}

.header__price-value-important {
  font-size: var(--font-size-s);
  font-weight: var(--font-weight-bold);
}

.header__account {
  flex-grow: 1;
  text-align: right;
  padding-right: var(--space-s);
}

.header__account-info,
.header__account-balances,
.header__account-balance:not(:first-child) {
  &:before {
    content: '|';
    margin-right: var(--space-xxs);
    padding-left: var(--space-xxs);
    color: var(--color-grey);
  }
}

.header__account-address {
  color: var(--color-grey);
}

.header__account-info {
  color: var(--color-loss)
}
