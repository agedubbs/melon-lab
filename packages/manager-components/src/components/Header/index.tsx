import React, { StatelessComponent } from 'react';
import Icon from '~/blocks/Icon';
import styles from './styles.css';

export interface HeaderProps {}

export const Header: StatelessComponent<HeaderProps> = ({ children }) => {
  return (
    <div className="header">
      <style jsx>{styles}</style>
      {children}
      <div className="header__logo">
        <Icon width="25px" height="25px" name="logos_default" />
      </div>
      <div className="header__name">Peter-Fund</div>
      <div className="header__price">
        <span className="header__price-desc">Share price:</span>
        <span className="header__price-value">1.000 MLN-T-M/Share</span>
      </div>
      <div className="header__price">
        <span className="header__price-desc">AUM:</span>
        <span className="header__price-value">1.000 MLN-T-M/Share</span>
      </div>
      <div className="header__price">
        <span className="header__price-desc">Ranking:</span>
        <span className="header__price-value">
          <span className="header__price-value-important">188</span> out of 287 Melon
          Funds
        </span>
      </div>
      <div className="header__account">
        <span className="header__account-address">0xA5f0…2783</span>
        <span className="header__account-balances">
          <span className="header__account-balance">Ⓜ 8.0000</span>
          <span className="header__account-balance">Ξ 0.3543</span>
        </span>
        <span className="header__account-net">kovan</span>
        <span className="header__account-info">Price feed down</span>
      </div>
    </div>
  );
};

export default Header;
