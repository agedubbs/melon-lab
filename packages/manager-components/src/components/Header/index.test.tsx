import React from 'react';
import Header from './index';

describe('Header', () => {
  const defaultElement = <Header />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
