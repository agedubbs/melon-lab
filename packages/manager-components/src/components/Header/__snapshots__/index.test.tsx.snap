// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div
  className="jsx-__scopedHash header"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <div
    className="jsx-__scopedHash header__logo"
  >
    <Icon
      height="25px"
      name="logos_default"
      width="25px"
    />
  </div>
  <div
    className="jsx-__scopedHash header__name"
  >
    Peter-Fund
  </div>
  <div
    className="jsx-__scopedHash header__price"
  >
    <span
      className="jsx-__scopedHash header__price-desc"
    >
      Share price:
    </span>
    <span
      className="jsx-__scopedHash header__price-value"
    >
      1.000 MLN-T-M/Share
    </span>
  </div>
  <div
    className="jsx-__scopedHash header__price"
  >
    <span
      className="jsx-__scopedHash header__price-desc"
    >
      AUM:
    </span>
    <span
      className="jsx-__scopedHash header__price-value"
    >
      1.000 MLN-T-M/Share
    </span>
  </div>
  <div
    className="jsx-__scopedHash header__price"
  >
    <span
      className="jsx-__scopedHash header__price-desc"
    >
      Ranking:
    </span>
    <span
      className="jsx-__scopedHash header__price-value"
    >
      <span
        className="jsx-__scopedHash header__price-value-important"
      >
        188
      </span>
       out of 287 Melon Funds
    </span>
  </div>
  <div
    className="jsx-__scopedHash header__account"
  >
    <span
      className="jsx-__scopedHash header__account-address"
    >
      0xA5f0…2783
    </span>
    <span
      className="jsx-__scopedHash header__account-balances"
    >
      <span
        className="jsx-__scopedHash header__account-balance"
      >
        Ⓜ 8.0000
      </span>
      <span
        className="jsx-__scopedHash header__account-balance"
      >
        Ξ 0.3543
      </span>
    </span>
    <span
      className="jsx-__scopedHash header__account-net"
    >
      kovan
    </span>
    <span
      className="jsx-__scopedHash header__account-info"
    >
      Price feed down
    </span>
  </div>
</div>
`;
